File: dotnet/src/view/mxStyleRegistry.cs
// Copyright (c) 2007-2008, Gaudenz Alder
using System;
using System.Diagnostics;
using System.Collections;
using System.Collections.Generic;
using System.Text;

namespace com.mxgraph
{
    /// <summary>
    /// Static class that acts as a global registry for codecs. See mxCodec for
    /// an example of using this class.
    /// </summary>
    public class mxStyleRegistry
    {

        /// <summary>
        /// Maps from strings to objects.
        /// </summary>
        protected static Dictionary<string, Object> values = new Dictionary<string, Object>();

        // Registers the known codecs and package names
        static mxStyleRegistry()
        {
            PutValue(mxConstants.EDGESTYLE_ELBOW, mxEdgeStyle.ElbowConnector);
            PutValue(mxConstants.EDGESTYLE_ENTITY_RELATION, mxEdgeStyle.EntityRelation);
            PutValue(mxConstants.EDGESTYLE_LOOP, mxEdgeStyle.Loop);
            PutValue(mxConstants.EDGESTYLE_SIDETOSIDE, mxEdgeStyle.SideToSide);
            PutValue(mxConstants.EDGESTYLE_TOPTOBOTTOM, mxEdgeStyle.TopToBottom);

            PutValue(mxConstants.PERIMETER_ELLIPSE, mxPerimeter.EllipsePerimeter);
            PutValue(mxConstants.PERIMETER_RECTANGLE, mxPerimeter.RectanglePerimeter);
            PutValue(mxConstants.PERIMETER_RHOMBUS, mxPerimeter.RhombusPerimeter);
            PutValue(mxConstants.PERIMETER_TRIANGLE, mxPerimeter.TrianglePerimeter);
        }

        /// <summary>
        /// Puts the given object into the registry under the given name.
        /// </summary>
        public static void PutValue(String name, Object value)
        {
            values[name] = value;
        }

        /// <summary>
        /// Returns the value associated with the given name.
        /// </summary>
        public static Object GetValue(String name)
        {
            if (values.ContainsKey(name))
            {
                return values[name];
            }

            return null;
        }

        /// <summary>
        /// Returns the value associated with the given name.
        /// </summary>
        public static String GetName(Object value)
        {
            foreach (String key in values.Keys)
            {
                if (values[key] == value)
                {
                    return key;
                }
            }

            return null;
        }

    }

}

File: php/src/view/mxStyleRegistry.php
<?php
/**
 * Copyright (c) 2006-2013, Gaudenz Alder
 */
class mxStyleRegistry
{

	/**
	 * Class: mxStyleRegistry
	 *
	 * Singleton class that acts as a global converter from string to object values
	 * in a style. This is currently only used to perimeters and edge styles.
	 * 
	 * Variable: values
	 *
	 * Maps from strings to objects.
	 */
	public static $values = array();

	/**
	 * Function: putValue
	 *
	 * Puts the given object into the registry under the given name.
	 */
	static function putValue($name, $value)
	{
		mxStyleRegistry::$values[$name] = $value;
	}

	/**
	 * Function: getValue
	 *
	 * Returns the value associated with the given name.
	 */
	static function getValue($name)
	{
		return (isset(mxStyleRegistry::$values[$name])) ? mxStyleRegistry::$values[$name] : null;
	}

	/**
	 * Function: getName
	 * 
	 * Returns the name for the given value.
	 */
	static function getName($value)
	{
		foreach (mxStyleRegistry::$values as $key => $val)
		{
			if ($value === $val)
			{
				return $key;
			}
		}
		
		return null;
	}

}

mxStyleRegistry::putValue(mxConstants::$EDGESTYLE_ELBOW, mxEdgeStyle::$ElbowConnector);
mxStyleRegistry::putValue(mxConstants::$EDGESTYLE_ENTITY_RELATION, mxEdgeStyle::$EntityRelation);
mxStyleRegistry::putValue(mxConstants::$EDGESTYLE_LOOP, mxEdgeStyle::$Loop);
mxStyleRegistry::putValue(mxConstants::$EDGESTYLE_SIDETOSIDE, mxEdgeStyle::$SideToSide);
mxStyleRegistry::putValue(mxConstants::$EDGESTYLE_TOPTOBOTTOM, mxEdgeStyle::$TopToBottom);

mxStyleRegistry::putValue(mxConstants::$PERIMETER_ELLIPSE, mxPerimeter::$EllipsePerimeter);
mxStyleRegistry::putValue(mxConstants::$PERIMETER_RECTANGLE, mxPerimeter::$RectanglePerimeter);
mxStyleRegistry::putValue(mxConstants::$PERIMETER_RHOMBUS, mxPerimeter::$RhombusPerimeter);
mxStyleRegistry::putValue(mxConstants::$PERIMETER_TRIANGLE, mxPerimeter::$TrianglePerimeter);
?>
